src/configuration/read.ts

```diff
@@ -0,0 +1,254 @@
+/**
+ * @license
+ * Copyright 2013 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as fs from "fs";
+import * as yaml from "js-yaml";
+import * as os from "os";
+import * as path from "path";
+import * as resolve from "resolve";
+import { FatalError, showWarningOnce } from "../error";
+import { arrayify, stripComments } from "../utils";
+import { IConfigurationFile } from "./configurationFile";
+
+const BUILT_IN_CONFIG = /^tslint:(.*)$/;
+
+// Note: eslint prefers yaml over json, while tslint prefers json over yaml
+// for backward-compatibility.
+export const JSON_CONFIG_FILENAME = "tslint.json";
+/** @deprecated use `JSON_CONFIG_FILENAME` or `CONFIG_FILENAMES` instead. */
+export const CONFIG_FILENAME = JSON_CONFIG_FILENAME;
+export const CONFIG_FILENAMES = [JSON_CONFIG_FILENAME, "tslint.yaml", "tslint.yml"];
+
+export interface RawConfigFile {
+    extends?: string | string[];
+    linterOptions?: IConfigurationFile["linterOptions"];
+    overrides?: Array<{
```

Comment:
These `Raw*` types were moved here and the `overrides?` key was added.  The functions in this file are unmodified.